nix/commands.nix

```diff
@@ -0,0 +1,360 @@
+{pkgs}:
+  let
+    dist_dir         = "./build";
+    node_modules_bin = "./node_modules/.bin";
+    src_dir				 	 = "./src";
+
+    ###
+
+    config					= "default";
+    fission_cmd     = "fission";
+    workbox_config 	= "workbox.config.cjs";
+
+    ###
+
+    devd            = "#{pkgs.devd}/bin/devd";
+    elm             = "${pkgs.elmPackages.elm}/bin/elm";
+    elm-format      = "${pkgs.elmPackages.elm-format}/bin/elm-format";
+    elm-live        = "${pkgs.elmPackages.elm-live}/bin/elm-live";
+    node            = "${pkgs.nodejs-14_x}/bin/node";
+    pnpm            = "${pkgs.nodePackages.pnpm}/bin/pnpm";
+    pnpx            = "${pkgs.nodePackages.pnpm}/bin/pnpx";
+    watchexec       = "${pkgs.watchexec}/bin/watchexec";
+
+    ###
+
+    task = name: body:
+      let
+        package =
+          pkgs.writeScriptBin name ''
+            #!${pkgs.stdenv.shell}
+            echo "⚙️  Running ${name}..."
+            ${body}
+          '';
+
+        bin = "${package}/bin/${name}";
+      in
+        { package = package;
+          bin     = bin;
+        };
+
+    ##
+
+    run = task "run" "${dev-server.bin} & ${watch.bin}";
```

Comment:
If I'm reading this right, this replaces just running `just` ... I wonder if we should try for some level of consistency here? (a la `npm start`).

Reply:
You mean rename this one to `start`?

Reply:
Actually, the other nice thing is that we can package these commands up and distribute them for all of our Elm projects via an `import`

Reply:
I don't have strong feelings `run` vs `start` but more should we try to have some consistency across projects (again so you drop in and know to just type `run` and away you go). 

but, yeah packaging them up as a base import is even better 🎉

Reply:
Okay, cool cool. I'll merge for now, and we can extract when the next person can't sleep and needs a project 😛 

Also thinking on this more... `run` is maybe more general (good in this case) as a "default" action, since it may not "start" anything

Reply:
FWIW, `start` is a *tiny* bit special in `npm`, as it's one of the scripts you can run without writing `run`. I mean. You can run the `start` script via `npm start`, but (I think) you'd have to run `npm run run` to run a `run` script.
I hope that made sense :smile:

Reply:
@matheus23 total sense, yeah I wasn't very clear in my original comment. I meant *like* how npm has a "special" command "start" we could/should have a "special" command for our nix setups that means "run this app". I think @expede makes a good case that "run" might actually work better than "start" here- but i'm less worried about what the command is, just that it's predictable